src/daft-scan/src/scan_task_iters.rs

```diff
@@ -320,26 +287,35 @@ fn split_and_merge_pass(
         .iter()
         .all(|st| st.as_any().downcast_ref::<ScanTask>().is_some())
     {
-        // TODO(desmond): Here we downcast Arc<dyn ScanTaskLike> to Arc<ScanTask>. ScanTask and DummyScanTask (test only) are
-        // the only non-test implementer of ScanTaskLike. It might be possible to avoid the downcast by implementing merging
-        // at the trait level, but today that requires shifting around a non-trivial amount of code to avoid circular dependencies.
-        let iter: BoxScanTaskIter = Box::new(scan_tasks.as_ref().iter().map(|st| {
-            st.clone()
-                .as_any_arc()
-                .downcast::<ScanTask>()
-                .map_err(|e| DaftError::TypeError(format!("Expected Arc<ScanTask>, found {:?}", e)))
-        }));
-        let split_tasks = split_by_row_groups(
-            iter,
-            cfg.parquet_split_row_groups_max_files,
-            cfg.scan_tasks_min_size_bytes,
-            cfg.scan_tasks_max_size_bytes,
-        );
-        let merged_tasks = merge_by_sizes(split_tasks, pushdowns, cfg);
-        let scan_tasks: Vec<Arc<dyn ScanTaskLike>> = merged_tasks
-            .map(|st| st.map(|task| task as Arc<dyn ScanTaskLike>))
-            .collect::<DaftResult<Vec<_>>>()?;
-        Ok(Arc::new(scan_tasks))
+        // Selectively apply scantask passes according to information from the Plan (e.g. pushdowns)
+        //
+        // If there is a "small" limit, then this is likely an attempt by a user to perform quick visualization. We return
+        // the ScanTasks unmolested in this case, utilizing the limit Pushdowns at read-time to appropriately elide I/O per-file.
+        //
+        // Otherwise, we assume that the user intention is for a large materialization of data.
+        if pushdowns.limit.map_or(false, |limit| limit <= 1000) {
```

Comment:
This is a little sus to me, I'm considering maybe instead of this heuristic we should add more context into the Pushdown to indicate that this limit is an "eager" limit.

We did add this idea of "eagerness" a while ago while optimizing for `.show()` workloads, because it affects execution. Curious to hear others' thoughts here.